mp/src/game/server/momentum/mom_replay_entity.cpp

```diff
@@ -409,6 +393,22 @@ void CMomentumReplayGhostEntity::HandleGhostFirstPerson()
     }
 }
 
+void CMomentumReplayGhostEntity::SetGhostAngles(QAngle angles)
+{
+    if (m_pCurrentSpecPlayer->GetObserverMode() == OBS_MODE_IN_EYE)
+    {
+        HideGhost();
+    }
+    else
+    {
+        // we divide x angle (pitch) by 10 so the ghost doesn't look really stupid
+        angles.x /= GHOST_PITCH_REDUCTION_VALUE;
+        UnHideGhost();
```

Comment:
Why are we setting ghost visibility in a method for setting the ghost's eye angles? Where it was before in `HandleGhostFirstPerson` is fine, moving and combining the functionality like this here could cause some unintended effects if things randomly set ghost angles in the future.

Reply:
I wanted to extract that code into a function so I could use it at the very end of the replay when the ghost is stuck in place. Because before when the player changed his view to a third-person the ghost just wouldn't appear. Maybe I should rename the function to a more descriptive name.

Reply:
In that case I would pull the visibility out into its own `DetermineGhostVisibility()` and call that in the places you need to. It shouldn't embedded here.